From c8ccc904d82b6a92f2f54ac499a4b995dcea47f1 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Thu, 29 Mar 2012 15:37:51 +0800
Subject: [PATCH 257/486] ENGR00178034 generic build and emulator running.

pass build of generic build.
and running the image with emulator.

modify all SECOND_DISPLAY to FSL_IMX_DISPLAY

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 mx5x/libgralloc/Android.mk        |  2 +-
 mx5x/libgralloc/framebuffer.cpp   | 14 +++++++-------
 mx6/libgralloc/Android.mk         |  4 +++-
 mx6/libgralloc/framebuffer.cpp    | 14 +++++++-------
 mx6/libgralloc_wrapper/Android.mk |  2 +-
 5 files changed, 19 insertions(+), 17 deletions(-)

diff --git a/mx5x/libgralloc/Android.mk b/mx5x/libgralloc/Android.mk
index 9f6eddd..9fd0774 100755
--- a/mx5x/libgralloc/Android.mk
+++ b/mx5x/libgralloc/Android.mk
@@ -43,7 +43,7 @@ LOCAL_CFLAGS += -DFSL_EPDC_FB
 endif
 
 ifeq ($(HAVE_FSL_IMX_IPU),true)
-LOCAL_CFLAGS += -DSECOND_DISPLAY_SUPPORT
+LOCAL_CFLAGS += -DFSL_IMX_DISPLAY
 endif
 
 LOCAL_MODULE_TAGS := eng
diff --git a/mx5x/libgralloc/framebuffer.cpp b/mx5x/libgralloc/framebuffer.cpp
index 42c53ad..7df658e 100755
--- a/mx5x/libgralloc/framebuffer.cpp
+++ b/mx5x/libgralloc/framebuffer.cpp
@@ -74,7 +74,7 @@ struct fb_context_t {
     int partial_width[20];
     int partial_height[20];
 #endif
-#ifdef SECOND_DISPLAY_SUPPORT
+#ifdef FSL_IMX_DISPLAY
     bool sec_display_inited;
     int sec_fp;
     int sec_disp_w;
@@ -102,7 +102,7 @@ struct fb_context_t {
 static int nr_framebuffers;
 static int no_ipu = 0;
 
-#ifdef SECOND_DISPLAY_SUPPORT
+#ifdef FSL_IMX_DISPLAY
 #define MAX_SEC_DISP_WIDTH (1024)
 #define MAX_SEC_DISP_HEIGHT (1024)
 static int mapSecFrameBuffer(fb_context_t* ctx);
@@ -334,7 +334,7 @@ static int fb_setUpdateRect(struct framebuffer_device_t* dev,
     return 0;
 }
 
-#ifdef SECOND_DISPLAY_SUPPORT
+#ifdef FSL_IMX_DISPLAY
 static int fb_setSecRotation(struct framebuffer_device_t* dev,int secRotation)
 {
     fb_context_t* ctx = (fb_context_t*)dev;
@@ -391,7 +391,7 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
         m->info.activate = FB_ACTIVATE_VBL;
         m->info.yoffset = offset / m->finfo.line_length;
 
-        #ifdef SECOND_DISPLAY_SUPPORT
+        #ifdef FSL_IMX_DISPLAY
         //Check the prop rw.SECOND_DISPLAY_CONNECTED
         char value[PROPERTY_VALUE_MAX];
      
@@ -542,7 +542,7 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
             return -errno;
         }
 
-    #ifdef SECOND_DISPLAY_SUPPORT
+    #ifdef FSL_IMX_DISPLAY
         if(ctx->sec_display_inited) sem_wait(&ctx->sec_display_end);
     #endif
 
@@ -1127,7 +1127,7 @@ static int mapFrameBuffer(struct private_module_t* module)
     return err;
 }
 
-#ifdef SECOND_DISPLAY_SUPPORT
+#ifdef FSL_IMX_DISPLAY
 static int mapSecFrameBuffer(fb_context_t* ctx)
 {
     int retCode = 0;
@@ -1653,7 +1653,7 @@ int fb_device_open(hw_module_t const* module, const char* name,
         dev->device.setUpdateRect = fb_setUpdateRect;
         #endif
         dev->device.compositionComplete = fb_compositionComplete;
-        #ifdef SECOND_DISPLAY_SUPPORT
+        #ifdef FSL_IMX_DISPLAY
         dev->device.setSecRotation = fb_setSecRotation;
         #endif
 
diff --git a/mx6/libgralloc/Android.mk b/mx6/libgralloc/Android.mk
index fd7b928..db02da1 100755
--- a/mx6/libgralloc/Android.mk
+++ b/mx6/libgralloc/Android.mk
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+ifneq (,$(findstring imx,$(TARGET_BOARD_PLATFORM)))
 LOCAL_PATH := $(call my-dir)
 
 # HAL module implemenation, not prelinked and stored in
@@ -43,9 +44,10 @@ LOCAL_CFLAGS:= -DLOG_TAG=\"$(TARGET_BOARD_PLATFORM).gralloc\" -D_LINUX
 #endif
 
 ifeq ($(HAVE_FSL_IMX_IPU),true)
-LOCAL_CFLAGS += -DSECOND_DISPLAY_SUPPORT
+LOCAL_CFLAGS += -DFSL_IMX_DISPLAY
 endif
 
 LOCAL_MODULE_TAGS := eng
 
 include $(BUILD_SHARED_LIBRARY)
+endif
diff --git a/mx6/libgralloc/framebuffer.cpp b/mx6/libgralloc/framebuffer.cpp
index 1431675..37ab14e 100755
--- a/mx6/libgralloc/framebuffer.cpp
+++ b/mx6/libgralloc/framebuffer.cpp
@@ -74,7 +74,7 @@ struct fb_context_t {
     int partial_width[20];
     int partial_height[20];
 #endif
-#ifdef SECOND_DISPLAY_SUPPORT
+#ifdef FSL_IMX_DISPLAY
     bool sec_display_inited;
     int sec_fp;
     int sec_disp_w;
@@ -104,7 +104,7 @@ struct fb_context_t {
 static int nr_framebuffers;
 static int no_ipu = 0;
 
-#ifdef SECOND_DISPLAY_SUPPORT
+#ifdef FSL_IMX_DISPLAY
 #define MAX_SEC_DISP_WIDTH (1024)
 #define MAX_SEC_DISP_HEIGHT (1024)
 static int mapSecFrameBuffer(fb_context_t* ctx);
@@ -336,7 +336,7 @@ static int fb_setUpdateRect(struct framebuffer_device_t* dev,
     return 0;
 }
 
-#ifdef SECOND_DISPLAY_SUPPORT
+#ifdef FSL_IMX_DISPLAY
 static int fb_setSecRotation(struct framebuffer_device_t* dev,int secRotation)
 {
     fb_context_t* ctx = (fb_context_t*)dev;
@@ -393,7 +393,7 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
         m->info.activate = FB_ACTIVATE_VBL;
         m->info.yoffset = offset / m->finfo.line_length;
 
-        #ifdef SECOND_DISPLAY_SUPPORT
+        #ifdef FSL_IMX_DISPLAY
         //Check the prop rw.SECOND_DISPLAY_CONNECTED
         char value[PROPERTY_VALUE_MAX];
      
@@ -467,7 +467,7 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
             return -errno;
         }
 
-    #ifdef SECOND_DISPLAY_SUPPORT
+    #ifdef FSL_IMX_DISPLAY
         if(ctx->sec_display_inited) sem_wait(&ctx->sec_display_end);
     #endif
 
@@ -1051,7 +1051,7 @@ static int mapFrameBuffer(struct private_module_t* module)
     return err;
 }
 
-#ifdef SECOND_DISPLAY_SUPPORT
+#ifdef FSL_IMX_DISPLAY
 static int mapSecFrameBuffer(fb_context_t* ctx)
 {
     int retCode = 0;
@@ -1497,7 +1497,7 @@ int fb_device_open(hw_module_t const* module, const char* name,
         dev->device.setUpdateRect = fb_setUpdateRect;
         #endif
         dev->device.compositionComplete = fb_compositionComplete;
-        #ifdef SECOND_DISPLAY_SUPPORT
+        #ifdef FSL_IMX_DISPLAY
         dev->device.setSecRotation = fb_setSecRotation;
         #endif
 
diff --git a/mx6/libgralloc_wrapper/Android.mk b/mx6/libgralloc_wrapper/Android.mk
index 58c3e32..fea7dcc 100755
--- a/mx6/libgralloc_wrapper/Android.mk
+++ b/mx6/libgralloc_wrapper/Android.mk
@@ -42,7 +42,7 @@ LOCAL_CFLAGS:= -DLOG_TAG=\"$(TARGET_BOARD_PLATFORM).gralloc\" -D_LINUX
 #endif
 
 ifeq ($(HAVE_FSL_IMX_IPU),true)
-LOCAL_CFLAGS += -DSECOND_DISPLAY_SUPPORT
+LOCAL_CFLAGS += -DFSL_IMX_DISPLAY
 endif
 
 LOCAL_MODULE_TAGS := eng
-- 
1.8.0

